lib/phoenix/socket/transport.ex

```diff
@@ -365,33 +365,40 @@ defmodule Phoenix.Socket.Transport do
     end)
   end
 
-  defp parse_origin(origin) do
+  defp parse_origin("http://*." <> orig),  do: parse_origin("http://" <> orig, :wildcard)
```

Comment:
In the previous implementation we already allowed `http://`, `https://` and `//`. I would rather support those values than introduce something that looks like a regular expression but works only in very specific conditions. It would also fit with `//foo.com/`, which we already support and is documented.

In other words, I think we can get rid of the whole `:static` and `:wildcard` thing and just use the values I showed above. The test suit additions are great though. :)